### DISCUSSION TITLE
DateTime type

### DISCUSSION THREAD

**Discussion by Sparclex** [spatie/typescript-transformer#53] [Q&A]
Hello 

I would like to transform every subclass of DateTimeInterface to `Brand<string, "DateTimeString">` in typescript. 

Given I have a class like this:

```php
class LastLogin extends Data 
{
    public function __construct(public CarbonImmutable $date) {}
}

```

it should be transformed to

```typescript
type LastLogin = {
   date: Brand<string, "DateTimeString">
}
```

But the collector I wrote does not seem to pick it up: 

```php
class DateTimeCollector extends Collector
{
    public function getTransformedType(ReflectionClass $class): ?TransformedType
    {
        if (! $class->isSubclassOf(DateTimeInterface::class)) {
            return null;
        }

        return (new DateTimeTransformer())->transform($class, $class->getShortName());
    }
}
class DateTimeTransformer implements Transformer
{
    public function transform(ReflectionClass $class, string $name): ?TransformedType
    {
        if (! $class->isSubclassOf(DateTimeInterface::class)) {
            return null;
        }

        return TransformedType::create(
            $class,
            'DateTimeString',
            'Brand<string, "DateTimeString">'
        );
    }
}

```

How do I need to write the collector in this case?

Thanks for the help!

**Comment by rubenvanassche** [comment]
Take a look at the default type replacements: https://github.com/spatie/typescript-transformer/blob/main/src/TypeScriptTransformerConfig.php#L72

If you're using Laravel, these are set within the config file. I think it should be possible over there to set the correct type.
